msal/src/main/java/com/microsoft/identity/client/internal/controllers/LocalMSALController.java

```diff
@@ -463,14 +468,224 @@ public boolean removeCurrentAccount(RemoveAccountCommandParameters parameters) t
+
+        // Loop to send multiple requests checking for token
+        while (errorCode.equals("authorization_pending")) {
+            errorCode = ""; // Reset error code
+
+            // Execute Token Request
+            tokenResult = oAuth2Strategy.requestToken(tokenRequest);
+
+            if (tokenResult.getErrorResponse() != null) {
+                errorCode = tokenResult.getErrorResponse().getError();
+            }
+
+            if (errorCode.equals("authorization_pending")) {
+                // interval is passed through params
+                Thread.sleep(interval);
+            }
+        }
+
+        // Assign token result
+        acquireTokenResult.setTokenResult(tokenResult);
+
+        // Validate request success, may throw MsalServiceException
+        validateServiceResult(tokenResult);
+
+        // If the token is valid, save it into token cache
+        final List<ICacheRecord> records = saveTokens(
+                oAuth2Strategy,
+                mAuthorizationRequest,
+                acquireTokenResult.getTokenResult().getTokenResponse(),
+                parameters.getOAuth2TokenCache()
+        );
+
+        // Once the token is stored, fetch and assign the authentication result
+        final ICacheRecord newestRecord = records.get(0);
+        acquireTokenResult.setLocalAuthenticationResult(
+                new LocalAuthenticationResult(
+                        finalizeCacheRecordForResult(
+                                newestRecord,
+                                parameters.getAuthenticationScheme()
+                        ),
+                        records,
+                        SdkType.MSAL,
+                        false
+                )
+        );
+
+        logResult(TAG, tokenResult);
+
+        // End telemetry with LOCAL_DEVICE_CODE_FLOW_POLLING
+        Telemetry.emit(
+                new ApiEndEvent()
+                        .putResult(acquireTokenResult)
+                        .putApiId(TelemetryEventStrings.Api.LOCAL_DEVICE_CODE_FLOW_POLLING)
+        );
+
+        return acquireTokenResult;
+    }
+
+    /**
+     * Helper method to check if a result object is valid (was a success). If not, an exception will be generated and thrown.
+     * This method is called in both parts of the DCF protocol.
+     * @param result result object to be checked
+     * @throws MsalServiceException MsalServiceException object reflecting error code returned by the result
+     */
+    private void validateServiceResult(@NonNull IResult result) throws MsalServiceException {
+        // If result was unsuccessful, create an exception
+        if (!result.getSuccess()) {
+            // Create ServiceException object
+            ServiceException serviceException = createServiceExceptionForDeviceCodeFlow(result.getErrorResponse());
+
+            // Convert ServiceException to MsalServiceException, then throw
+            throw new MsalServiceException(
```

Comment:
Why are we creating a ServiceException when we want to throw `MsalServiceException`? Why not create the latter directly?

Reply:
I was looking at some other code and it seems like a `ServiceException` is part of creating an `MsalServiceExeption` object. the `ServiceException` is included as the "throwable". I figured this would be useful somepoint down the line but it would make sense to just create an `MsalServiceException` right away.